noir-projects/aztec-nr/aztec/src/messages/encoding.nr

```diff
@@ -1,9 +1,30 @@
 // TODO(#12750): don't make these values assume we're using AES.
-use crate::{
-    messages::encryption::log_encryption::PRIVATE_LOG_PLAINTEXT_SIZE_IN_FIELDS, utils::array,
-};
-
-pub global MAX_MESSAGE_LEN: u32 = PRIVATE_LOG_PLAINTEXT_SIZE_IN_FIELDS;
+use crate::utils::array;
+use protocol_types::constants::PRIVATE_LOG_CIPHERTEXT_LEN;
+
+// We reassign to the constant here to communicate the distinction between a log and a message. In Aztec.nr, unlike in
+// protocol circuits, we have a concept of a message that can be emitted either as a private log or as an offchain
+// message. Message is a piece of data that is to be eventually delivered to a contract via the `process_message(...)`
+// utility function function that is inject by the #[aztec] macro.
+pub global MESSAGE_CIPHERTEXT_LEN: u32 = PRIVATE_LOG_CIPHERTEXT_LEN;
+
+// TODO(#12750): The global variables below should not be here as they are AES128 specific.
+// ciphertext_length (2) + 14 bytes pkcs#7 AES padding.
+pub(crate) global HEADER_CIPHERTEXT_SIZE_IN_BYTES: u32 = 16;
+
+pub global EPH_PK_X_SIZE_IN_FIELDS: u32 = 1;
+pub global EPH_PK_SIGN_BYTE_SIZE_IN_BYTES: u32 = 1;
+
+// (17 - 1) * 31 - 16 - 1 = 479
+global MESSAGE_PLAINTEXT_SIZE_IN_BYTES: u32 = (MESSAGE_CIPHERTEXT_LEN - EPH_PK_X_SIZE_IN_FIELDS)
+    * 31
+    - HEADER_CIPHERTEXT_SIZE_IN_BYTES
+    - EPH_PK_SIGN_BYTE_SIZE_IN_BYTES;
+// Each field of the original note log was serialized to 32 bytes. Below we convert the bytes back to fields.
+// 479 / 32 = 15
+pub global MESSAGE_PLAINTEXT_SIZE_IN_FIELDS: u32 = MESSAGE_PLAINTEXT_SIZE_IN_BYTES / 32;
```

Comment:
These values were moved in here from `log_encryption.nr` and renamed to reflect the message naming. This allowed me to hide the `MESSAGE_PLAINTEXT_SIZE_IN_BYTES` which I think is a good sign (meaning it was only used in this file and previously imported from `log_encryption.nr`).

Think this file being called `encoding.nr` makes this quite a natural place for these constants but if the reviewer have a better idea where to place these I am fine with reworking this.

The `HEADER_CIPHERTEXT_SIZE_IN_BYTES` being here is ugly but having that in `log_encryption.nr` was ugly as well so I think I didn't worsen it.